### PR TITLE
adding API, and integrating all modules

### PR DESCRIPTION
**Pull request by Manik-Jain**

### CODE REVIEW

File: kafka/eventType.js [hunk 1]
```diff
@@ -23,6 +23,42 @@ export default avro.Type.forSchema({
     {
       name: 'name',
       type: 'string',
+    },
```
Comment (dmitri-safine) [inline kafka/eventType.js:26]: So do we treat every field in the authenticity certificate as a different type of event in Kafka message? I don't think we should go that granular. I'd rather see `certificate` as an event type, with all the fields there as JSON.

File: api/index.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
+import {
+    app,
+    properties
+} from './conf/conf.js'
+import StatusCodes from 'http-status-codes'
+import express from 'express'
+import got from 'got'
+import CardanoMessageProducer from '../kafka/producer/index.js'
```
Comment (dmitri-safine) [inline api/index.js:8]: I'd rather do `CardanoMessageProducer` as generic module and use `KafkaMessageProducer` as one of it's implementations. Can we follow the dependency injection pattern here? https://www.npmjs.com/package/dependency-injection-es6

**Review by dmitri-safine — COMMENTED** [review]
Looks great for the most part, just a few comments and questions.

File: cmds/index.js [hunk 3]
```diff
@@ -2,5 +2,8 @@ import CardanoCliExecutor from './CardanoCliExecutor.js'
 import Commands from './commands.js'
 
 let cardanoCli = new CardanoCliExecutor();
-cardanoCli.execute(Commands.Commands.LIST_FILES);
+//cardanoCli.execute(Commands.Commands.LIST_FILES);
+
+let t = `curl -H 'project_id: yIiASEfGiDNeVjPhydXIjRX8Y5Mdw2PA' https://cardano-testnet.blockfrost.io/api/v0/txs/e13b1ec55e64d0807895ba81435e4275839dd57b5bb915d924b58eb3e6b1f390/metadata`;
```
[the remark this answers is not in the document]
Comment (Manik-Jain) [reply cmds/index.js:7]: yes. this was a test. I'll remove this

File: kafka/eventType.js (hunk 1 above)
Comment (Manik-Jain) [reply kafka/eventType.js:26]: yes, it is the same way as you've mentioned. Kafka receives a single JSON message with all the fields. The intention of this file is to restrict the Kafka to have a specific message type. Like we have the notion of classes in other languages, like Java.

File: api/index.js (hunk 2 above)
Comment (Manik-Jain) [reply api/index.js:8]: And, Yes we can use Dependency Injection all across the project. That's a nice idea! 
I'll create a task for that and create another Pull Request for its implementation.

File: cmds/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (Manik-Jain) [reply cmds/index.js]: done

File: api/index.js (hunk 2 above)
Comment (Manik-Jain) [reply api/index.js:8]: Task created for this : https://github.com/adamantas/cardano-fluree/issues/19

File: kafka/eventType.js (hunk 1 above)
Comment (dmitri-safine) [reply kafka/eventType.js:26]: Great! Just wanted to make sure.

**Pull request merged**